tests/test_mbed_targets.py

```diff
@@ -238,6 +241,16 @@ def test_json_dump_from_raw_and_fitered_data(self, mocked_get_offline_target_dat
 
         self.assertEqual(json_str_from_filtered, json.dumps([t1_filt.__dict__, t2_filt.__dict__], indent=4))
 
+    @mock.patch("mbed_targets._internal.target_database.get_offline_target_data")
+    def test_handles_database_errors(self, mocked_get_offline_target_data, mocked_get_online_target_data):
```

Comment:
I think this should be separated into two separate tests (and potentially TestCases). Seems to me we're trying too hard to fit all tests under `TestMbedTargets` umbrella.

Reply:
Do you mind if that's done in another PR? After this and the CI fix are in I would like to rationalise the tests as I think they're nonsensical in some cases.

Reply:
Aye.